standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java

```diff
@@ -247,8 +252,8 @@ private void startLocalSegmentStore(int segmentStoreId) throws Exception {
                         .with(ServiceConfig.LISTENING_PORT, this.segmentStorePorts[segmentStoreId])
                         .with(ServiceConfig.CLUSTER_NAME, this.clusterName)
                         .with(ServiceConfig.ENABLE_TLS, this.enableTls)
-                        .with(ServiceConfig.KEY_FILE, this.keyFile)
-                        .with(ServiceConfig.CERT_FILE, this.certFile)
+                        .with(ServiceConfig.KEY_FILE, this.keyFile == null ? "" : this.keyFile)
```

Comment:
I'm still not sure why you don't want to change the builder to do exactly that. To be precise:

- If I have `.with(ServiceConfig.KEY_FILE, value)` in my build, then use the value
- If no value is passed for `ServiceConfig.KEY_FILE`, then use ""

Reply:
What does "no value" mean? And why replace it with the arbitrary empty string?

`ServiceConfig.KEY_FILE` has a default value (check the property). If no special value needs to be passed, then do not invoke the `.with(ServiceConfig.KEY_FILE, ...)` on this object. 

There is no reason why we need to change the `ConfigBuilder` class, which is shared by all the configurations in Pravega.

Reply:
The approach recommended by @andreipaduroiu was the one I started with https://github.com/pravega/pravega/pull/2535/commits/0d20611c1474e719ab21e0ddf05a81596f538b85. I think we should go with that.

Reply:
My earlier feedback had to do with supporting a fluent style (as opposed to the ugly multi-statement approach seen in [0d20611](https://github.com/pravega/pravega/commit/0d20611c1474e719ab21e0ddf05a81596f538b85)).

A possible compromise would be to add a new method to [`ConfigBuilder`](https://github.com/pravega/pravega/blob/master/common/src/main/java/io/pravega/common/util/ConfigBuilder.java):
```java
public <V> ConfigBuilder<T> with(Property<V> property, Optional<V> value) {
    if(value.isPresent()) {
        with(property, value.get());
    }
    return this;
}
```

In that way, one could use a fluent style with nullable values (e.g. keyfile):
```
builder
    .with(ServiceConfig.KEY_FILE, Optional.ofNullable(this.keyFile))
    ...
```

Reply:
@andreipaduroiu this PR initially had this:

```
if (this.enableTls) {
            serviceConfigConfigBuilder = serviceConfigConfigBuilder.with(ServiceConfig.KEY_FILE, this.keyFile)
                    .with(ServiceConfig.CERT_FILE, this.certFile);
 }
```
and now it has been changed to:

```
.with(ServiceConfig.KEY_FILE, this.keyFile == null ? "" : this.keyFile)
```

The behavior you are describing is precisely what I'd like to see.